config/v1alpha1/types_insights.go

```diff
@@ -48,17 +47,82 @@ type GatherConfig struct {
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	DisabledGatherers []DisabledGatherer `json:"disabledGatherers"`
+	// storageSpec is an optional field that allows user to define persistent storage for on-demand gathering
+	// jobs to store the Insights data archive.
+	// If omitted, the gathering job will use ephemeral storage.
+	// +optional
+	StorageSpec *StorageSpec `json:"storageSpec,omitempty"`
+}
+
+// disabledGatherer is a string that represents a gatherer that should be disabled
+// +kubebuilder:validation:MaxLength=256
+// +kubebuilder:validation:XValidation:rule=`self.matches("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$")`,message=`disabledGatherer must be in the format of {gatherer}/{function} where the gatherer and function are lowercase strings that may include underscores (_) and are separated by a forward slash (/) if the function is provided`
+type DisabledGatherer string
+
+// storageSpec provides persistent storage configuration options for on-demand gathering jobs.
+// If the type is set to PersistentVolumeClaim, then the PersistentVolume must be defined.
+// If the type is set to Ephemeral, then the PersistentVolume must not be defined.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'PersistentVolumeClaim' ?  has(self.persistentVolume) : !has(self.persistentVolume)",message="persistentVolume is required when type is PersistentVolumeClaim, and forbidden otherwise"
+type StorageSpec struct {
+	// type is a required field that specifies the type of storage that will be used to store the Insights data archive.
+	// Valid values are "PersistentVolumeClaim" and "Ephemeral".
+	// +required
+	Type StorageType `json:"type"`
+	// persistentVolume is an optional field that specifies the PersistentVolume that will be used to store the Insights data archive.
+	// The PersistentVolume must be created in the openshift-insights namespace.
+	// +optional
+	PersistentVolume *PersistentVolumeConfig `json:"persistentVolume,omitempty"`
+}
+
+// storageType declares valid storage types
+// +kubebuilder:validation:Enum=PersistentVolumeClaim;Ephemeral
+type StorageType string
+
+const (
+	// StorageTypePersistentVolumeClaim storage type
+	StorageTypePersistentVolumeClaim StorageType = "PersistentVolumeClaim"
+	// StorageTypeEphemeral storage type
+	StorageTypeEphemeral StorageType = "Ephemeral"
+)
+
+// persistentVolumeConfig provides configuration options for PersistentVolume storage.
+type PersistentVolumeConfig struct {
+	// persistentVolumeClaim is a required field that specifies the configuration of the PersistentVolumeClaim that will
+	// be used to store the Insights data archive. The PersistentVolumeClaim must be created in the openshift-insights namespace.
+	// +required
+	PersistentVolumeClaim PersistentVolumeClaimReference `json:"persistentVolumeClaim"`
```

Comment:
This field will look like

```
storageSpec:
  type: PersistentVolume
  persistentVolume:
    persistentVolumeClaim:
      name: ...
```
Do we need to repeat `peristentVolume` again once we are inside that scope?
```
storageSpec:
  type: PersistentVolume
  persistentVolume:
    claim:
      name: ...
```
Were the two options discussed already?

Reply:
AFAIK it was not discussed and I don't mind changing the `persistentVolumeClaim` to `claim`